src/components/EditableField/styles/EditableField.Input.css.ts

```diff
@@ -0,0 +1,314 @@
+import styled from '../../styled/index'
+import baseStyles from '../../../styles/resets/baseStyles.css'
+import { getColor } from '../../../styles/utilities/color'
+
+export const EditableFieldInputUI = styled('div')`
+  ${baseStyles};
+  position: relative;
+  height: 25px;
+  margin-bottom: 10px;
+
+  &:hover .EditableField__actions {
+    opacity: 1;
+    cursor: pointer;
+  }
+
+  &.is-active:hover .EditableField__actions {
+    display: none;
+    cursor: initial;
+  }
+`
+
+export const InteractiveContentUI = styled('div')`
+  display: flex;
+  height: 26px;
+  width: 100%;
+  max-width: 100%;
+  position: absolute;
+  top: 0;
+  left: 0;
+  z-index: 1;
+  border-bottom: 1px solid transparent;
+
+  &:hover {
+    cursor: pointer;
+    border-bottom: 1px dashed #c6d0d8;
+  }
+
+  .is-disabled &:hover {
+    cursor: initial;
+  }
+
+  .is-active & {
+    pointer-events: none;
+    z-index: 2;
+    border-bottom-color: transparent !important;
+  }
```

Comment:
Just a question for you @tinkertrain, what do you think about using props inside the CSS instead of class names?

Reply:
I think it depends on the situation, I like classes because:

- you can override the styles externally if you need to with minimal effort
- Is more familiar with more developers
- Styled components will create new classes for each variation, which I'm guessing with no real evidence that it might not be the most performant (kind of like @extend in sass? maybe I'm way off on this one though)

I like props because of their dynamism and cool things you can do, sometimes they are the right tool for the job, there's no easy way to achieve what line EditableField.Input.css.ts:215 does with classes, and I think that is awesome

Reply:
Great points @tinkertrain! I guess with the way we've been using classes too, they help identify what state something is in, like `is-disabled`, for example so that's helpful for testing. Cheers!